src/decisionengine/framework/engine/Workers.py

```diff
@@ -47,38 +48,70 @@ def get_consumes(self):
         return self.task_manager.get_consumes()
 
     def run(self):
-        logger = logging.getLogger()
-        logger.setLevel(logging.WARNING)
+
+        myname = self.task_manager.name
+        myfilename = os.path.join(
+            os.path.dirname(self.logger_config["log_file"]), myname + ".log"
+        )
+
+        logger = structlog.getLogger(f"{myname}")
+
+        # setting a default value here. value from config file is set in call
+        # self.task_manager.set_loglevel_value after logger configuration is completed
+        logger.setLevel(logging.DEBUG)
         logger_rotate_by = self.logger_config.get("file_rotate_by", "size")
 
+        logconf = configDict.pylogconfig.copy()
```

Comment:
Can you switch this to `copy.deepcopy` (and add the `import copy`)?  Somehow I missed this one earlier....